[4.0] Breaking Changes (#1209)
telerik/blazor-docs: upgrade/breaking-changes/4-0-0 .md

---
title: 4.0.0
description: Handle the changes in the43.0.0 release of the Telerik UI for Blazor components.
page_title: Breaking Changes in 4.0.0

> slug: changes-in-4-0-0

Reviewer: Link the page from the main ("List") article.

Author: I will, thank you, forgot about that.

position: 1
---

# Breaking Changes in 4.0.0

> ## Common Changes

Reviewer: Currently there is a discrepancy in our approach with regard to identical changes in multiple components. The `Stretched` value is in the Common section. On the other hand, the `Shape` parameter is repeated in each component section.

Author: Agreed, moved the Shape parameter to the common changes.

> ### Supported .NET versions

Reviewer: ```suggestion
### Supported .NET Versions
```

Microsoft drops the support of `.NET Core 3.1.x` on December 13, 2022. From the 4.0.0 release, Telerik UI for Blazor will support `.NET 6` and newer versions.

> ### Stretch enum value

Reviewer: It may be easier for customers to recognize the old API member names, rather than the new ones. This will make the section look more relevant to the reader. What do you think?

```suggestion
### Stretched enum Value
```

* [Card Actions `Layout`]({%slug card-actions%}#layout)

> * [Dialog Action `Buttons Layout`]({%slug dialog-action-buttons%})

Reviewer: ```suggestion
* [Dialog Action `ButtonsLayout`]({%slug dialog-action-buttons%})
```

> * [Grid PopUp `Form Buttons Layout`]({%slug components/grid/editing/popup%}#edit-form-customization)

Reviewer: ```suggestion
* [Grid popup edit form `ButtonsLayout`]({%slug components/grid/editing/popup%}#edit-form-customization)
```

* [Scheduler PopUp `Form Buttons Layout`]({%slug scheduler-edit-popup-customization%}#edit-form-customization)

> * [TreeList PopUp `Form Buttons Layout`]({%slug treelist-editing-popup%}#edit-form-customization)

Reviewer: ```suggestion
* [TreeList popup edit form `ButtonsLayout`]({%slug treelist-editing-popup%}#edit-form-customization)
```

</code></pre>
            </td>
        </tr>
        <tr>
            <td style="vertical-align:top">
<pre><code>
&lt;GridPopupEditFormSettings ButtonsLayout="FormButtonsLayout.Stretched" /&gt;
</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
&lt;GridPopupEditFormSettings ButtonsLayout="FormButtonsLayout.Stretch" /&gt;
</code></pre>
            </td>
        </tr>
        <tr>
            <td style="vertical-align:top">
<pre><code>
&lt;SchedulerPopupEditFormSettings ButtonsLayout="FormButtonsLayout.Stretched" /&gt;
</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
&lt;SchedulerPopupEditFormSettings ButtonsLayout="FormButtonsLayout.Stretch" /&gt;
</code></pre>
            </td>
        </tr>
        <tr>
            <td style="vertical-align:top">
<pre><code>
&lt;TreeListPopupEditFormSettings ButtonsLayout="FormButtonsLayout.Stretched" /&gt;
</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
&lt;TreeListPopupEditFormSettings ButtonsLayout="FormButtonsLayout.Stretch" /&gt;
</code></pre>
            </td>
        </tr>
    </tbody>
</table>

### Async Method Suffix

> Added the `Async` suffix to the signature of asynchronous methods

Reviewer: "signature" is correct, but why not say it simpler? Do we expect other methods here, or should we focus on `SetState` only?

```suggestion
Added an `Async` suffix to the name of asynchronous methods.
```

>caption Method signatures in UI for Blazor up to version 3.8.0 and after version 4.0.0

<table>
    <thead><tr>
        <th>UI for Blazor 3.8.0</th>
        <th>UI for Blazor 4.0.0</th>
    </tr></thead>
    <tbody>
        <tr>
            <td style="vertical-align:top">
<pre><code>

> await TelerikGridReference.SetState(desiredState);

Reviewer: ```suggestion
// private TelerikGrid<GridModel> GridRef { get; set; }
await GridRef.SetState(desiredState);
```

</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
await TelerikGridReference.SetStateAsync(desiredState);
</code></pre>
            </td>
        </tr>
        <tr>
            <td style="vertical-align:top">
<pre><code>
await TelerikTreeListReference.SetState(desiredState);
</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
await TelerikTreeListReference.SetStateAsync(desiredState);
</code></pre>
            </td>
        </tr> 
    </tbody>
</table>

### Obsolete Parameters

> * Removed the `DefaultItem` parameter from the Telerik DropDownList.

Author: Nothing, we just remove it. When the actual removal happens we might suggest something.

> * Removed the `DefaultItem` parameter from the Telerik DropDownList.

Author: You are right, moved it to a DropDownList section

## Component Changes

### Button

- Removed the `Shape` parameter.

> - Changed the signature of the `Hidden` parameter to `Visible`.

Reviewer: ```suggestion
- Renamed the `Hidden` parameter to `Visible`.
```

>caption Visible parameter in UI for Blazor up to version 3.8.0 and after version 4.0.0

<table>
    <thead><tr>
        <th>UI for Blazor 3.8.0</th>
        <th>UI for Blazor 4.0.0</th>
    </tr></thead>
    <tbody>
        <tr>
            <td style="vertical-align:top">
<pre><code>
&lt;TelerikButton Hidden="false"&gt; Visible button &lt;TelerikButton /&gt;
</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
&lt;TelerikButton Visible="true"&gt; Visible button &lt;TelerikButton /&gt;
</code></pre>
            </td>
        </tr>
    </tbody>
</table>

### ButtonGroup

- Removed the `Shape` parameter.

### Drawer

- Changed the values in the `DrawerPosition` enum from `Left` and `Right` to `Start` and `End`.

>caption DrawerPosition enum values in UI for Blazor up to version 3.8.0 and after version 4.0.0

<table>
    <thead><tr>
        <th>UI for Blazor 3.8.0</th>
        <th>UI for Blazor 4.0.0</th>
    </tr></thead>
    <tbody>
        <tr>
            <td style="vertical-align:top">
<pre><code>

> &lt;TelerikDrawer Position="@DrawerPosition.Left/Right"&gt;

Reviewer: Let's use valid code snippets and count that our customers will fill in the blanks.

```suggestion
&lt;TelerikDrawer Position="@DrawerPosition.Left"&gt;
```

</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>

> &lt;TelerikDrawer Position="@DrawerPosition.Start/End"&gt;

Reviewer: ```suggestion
&lt;TelerikDrawer Position="@DrawerPosition.Start"&gt;
```

</code></pre>
            </td>
        </tr>
    </tbody>
</table>

### Form

- The Form columns can have different widths

### Grid

- Removed default filter descriptors in the state when the Grid `FilterMode` is set to `FilterMenu`.

> - Removed `Primary` parameter from the `<GridCommandButton>`.

Reviewer: ```suggestion
- Removed the `Primary` parameter from `<GridCommandButton>`.
```

            </td>
            <td style="vertical-align:top">
<pre><code>
&lt;TelerikTextArea Placeholder="The placeholder text" /&gt;
</code></pre>
            </td>
        </tr>
    </tbody>
</table>

### TextBox

- Changed the signature of the `PlaceHolder` parameter to `Placeholder`.

>caption Placeholder parameter in UI for Blazor up to version 3.8.0 and after version 4.0.0

<table>
    <thead><tr>
        <th>UI for Blazor 3.8.0</th>
        <th>UI for Blazor 4.0.0</th>
    </tr></thead>
    <tbody>
        <tr>
            <td style="vertical-align:top">
<pre><code>
&lt;TelerikTextBox PlaceHolder="The placeholder text" /&gt;
</code></pre>
            </td>
            <td style="vertical-align:top">
<pre><code>
&lt;TelerikTextBox Placeholder="The placeholder text" /&gt;
</code></pre>
            </td>
        </tr>
    </tbody>
</table>

### ToggleButton

- Removed the `Shape` parameter.

### Toolbar

- Changed the default value of the `Adaptive` parameter from `false` to `true`.

### TreeList

- Removed default filter descriptors in the state when the TreeList `FilterMode` is set to `FilterMenu`.
- Removed `Primary` parameter from the `<TreeListCommandButton>`.

> >important Any snippet created before `Telerik.UI.for.Blazor` version 4.0 (January 18, 2023) may need an update due to the above list of changes.

Reviewer: Some customers may find this sentence ambiguous. Are we talking specifically about REPL as in the 3.0 article? For regular production apps this should be common sense?
